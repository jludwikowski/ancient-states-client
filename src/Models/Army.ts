import {Entity} from './Entity';
import Leader from './Leader';
import UnitReference from "./UnitReference";

export default interface Army extends Entity {
    commander: Leader | null,
    units: UnitReference[]
    owner: number,
    position: string,
    status: string,
    orderDate?: string | null,
    targetId?: number | null,
}
