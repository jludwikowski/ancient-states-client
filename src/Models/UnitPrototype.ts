import {Entity} from './Entity';
import {UnitType} from './UnitType';

export default interface UnitPrototype extends Entity {
    baseTrainTime: number,
    baseDamage: number,
    activeTurns: number[],
    baseHealth: number,
    baseCost: number,
    type: UnitType,
    imageUrl: string
}
