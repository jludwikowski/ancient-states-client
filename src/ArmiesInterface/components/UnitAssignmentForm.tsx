import React from "react";
import Army from "../../Models/Army";
import UnitPrototype from "../../Models/UnitPrototype";
import UnitReference from "../../Models/UnitReference";
import {SingleUnitForm} from "./SingleUnitForm";

interface Props {
    army: Army,
    unitPrototypes: UnitPrototype[],
    getUnit(event:React.ChangeEvent<HTMLInputElement>): void,
}

export class UnitAssignmentForm extends React.PureComponent<Props,UnitReference[]> {

    public render() {
        return (
            <div className="input-group mb-3">
                {this.props.army.units ? (
                    Object.values(this.props.army.units).map((unit, index) => (
                        <SingleUnitForm key={index} unit={unit} getUnit={this.props.getUnit} unitPrototypes={this.props.unitPrototypes} />
                    ))) : null }
            </div>
        );
    }
}
