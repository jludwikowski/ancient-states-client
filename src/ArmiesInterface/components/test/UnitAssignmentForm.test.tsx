import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Army from "../../../Models/Army";
import UnitPrototype from "../../../Models/UnitPrototype";
import { UnitAssignmentForm } from '../UnitAssignmentForm';

describe('UnitAssignmentForm', () => {

    const army:Army = {
        id: 7,
        name: 'New Army',
        description: '',
        status: 'camp',
        orderDate: null,
        targetId: null,
        position: '0,0',
        owner: 1,
        commander: null,
        units: [
            {
                number: 9802,
                level: 1,
                baseUnit: 1,
            }
        ]
    };
    const baseUnits:UnitPrototype[] = [{
        id: 1,
        name: "Persian Spearman",
        description: "Poorly armed and armor eastern warrior",
        baseTrainTime: 50,
        baseDamage: 8,
        baseHealth: 30,
        baseCost: 40,
        activeTurns: [6, 7, 8, 9, 10, 11, 12],
        type: "Melee",
        imageUrl: "images/icons/icons8-military-base-100.png"
    }];

    const getUnit = (event:React.ChangeEvent<HTMLInputElement>) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<UnitAssignmentForm army={army} unitPrototypes={baseUnits} getUnit={getUnit} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<UnitAssignmentForm army={army} unitPrototypes={baseUnits} getUnit={getUnit} />);
        expect(component).toMatchSnapshot();
    });
});
