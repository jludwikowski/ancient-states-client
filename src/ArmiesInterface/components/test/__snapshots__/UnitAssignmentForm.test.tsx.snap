// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitAssignmentForm Should match snapshot 1`] = `
<div
  className="input-group mb-3"
>
  <SingleUnitForm
    getUnit={[Function]}
    key="0"
    unit={
      Object {
        "baseUnit": 1,
        "level": 1,
        "number": 9802,
      }
    }
    unitPrototypes={
      Array [
        Object {
          "activeTurns": Array [
            6,
            7,
            8,
            9,
            10,
            11,
            12,
          ],
          "baseCost": 40,
          "baseDamage": 8,
          "baseHealth": 30,
          "baseTrainTime": 50,
          "description": "Poorly armed and armor eastern warrior",
          "id": 1,
          "imageUrl": "images/icons/icons8-military-base-100.png",
          "name": "Persian Spearman",
          "type": "Melee",
        },
      ]
    }
  />
</div>
`;
